components/renku_data_services/storage_adapters/core.py

```diff
@@ -36,10 +37,57 @@ async def get_storage(self, id: str | None = None, project_id: str | None = None
             orms = res.scalars().all()
             return [orm.dump() for orm in orms]
 
+    async def get_storage_by_id(self, storage_id: str) -> models.CloudStorage:
+        """Get a single storage by id."""
+        async with self.session_maker() as session:
+            res = await session.execute(
+                select(schemas.CloudStorageORM).where(schemas.CloudStorageORM.storage_id == storage_id)
+            )
+            storage = res.one_or_none()
+
+            if storage is None:
+                raise errors.MissingResourceError(message=f"The storage with id '{storage_id}' cannot be found")
+            return storage[0].dump()
+
     async def insert_storage(self, storage: models.CloudStorage) -> models.CloudStorage:
         """Insert a new cloud storage entry."""
         orm = schemas.CloudStorageORM.load(storage)
         async with self.session_maker() as session:
             async with session.begin():
                 session.add(orm)
         return orm.dump()
+
+    async def update_storage(self, storage_id: str, **kwargs) -> models.CloudStorage:
+        """Update a cloud storage entry."""
+        async with self.session_maker() as session:
+            async with session.begin():
+                res = await session.execute(
+                    select(schemas.CloudStorageORM).where(schemas.CloudStorageORM.storage_id == storage_id)
+                )
+                storage = res.one_or_none()
+
+                if storage is None:
+                    raise errors.MissingResourceError(message=f"The storage with id '{storage_id}' cannot be found")
+                storage = storage[0]
+
+                for key, value in kwargs.items():
+                    setattr(storage, key, value)
+
+                if "configuration" in kwargs and "type" in kwargs["configuration"]:
+                    storage.storage_type = kwargs["configuration"]["type"]
```

Comment:
So the storage type is set both in `storage.storage_type` and `storage.configuration.type`? Is there a reason for this. Can we get away with having just one and not both. And if we have to have both we should check for consistency somewhere?